# 关节状态控制器 - 负责发布所有关节的状态
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50
  joints:
    - joint1
    - joint2
    - joint3
    - joint4
    - joint5
    - r_joint
    - r_in_joint
    - r_out_joint
    - l_joint
    - l_in_joint
    - l_out_joint


# 机械臂主体控制器 - 控制机械臂的5个主要关节
arm_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - joint1
    - joint2
    - joint3
    - joint4
    - joint5
  gains:
    joint1: {p: 40, d: 2, i: 1, i_clamp: 1}
    joint2: {p: 10, d: 2, i: 1, i_clamp: 1}
    joint3: {p: 2, d: 2, i: 1, i_clamp: 1}
    joint4: {p: 5, d: 1, i: 1, i_clamp: 1}
    joint5: {p: 5, d: 1, i: 1, i_clamp: 1}
  constraints:
    goal_time: 0.6
    stopped_velocity_tolerance: 0.05
    joint1: {trajectory: 0.1, goal: 0.1}
    joint2: {trajectory: 0.1, goal: 0.1}
    joint3: {trajectory: 0.1, goal: 0.1}
    joint4: {trajectory: 0.1, goal: 0.1}
    joint5: {trajectory: 0.1, goal: 0.1}
  stop_trajectory_duration: 0.5
  state_publish_rate: 125
  action_monitor_rate: 10

# 夹爪控制器
gripper_controller:
  type: effort_controllers/JointPositionController
  joint: r_joint
  pid: {p: 1, i: 0, d: 10}

